pkg/util/informermanager/single-cluster-manager.go

```diff
@@ -18,13 +18,21 @@ type SingleClusterInformerManager interface {
 	// The handler should not be nil.
 	ForResource(resource schema.GroupVersionResource, handler cache.ResourceEventHandler)
 
+	// ForResourceIndexer builds a dynamic shared informer indexers for 'resource' then add indexers into the informer.
+	// The informer for 'resource' will be created if not exist, but without any event handler.
+	ForResourceIndexer(resource schema.GroupVersionResource, indexers cache.Indexers) error
+
 	// IsHandlerExist checks if handler already added to a the informer that watches the 'resource'.
 	IsHandlerExist(resource schema.GroupVersionResource, handler cache.ResourceEventHandler) bool
 
 	// Lister returns a generic lister used to get 'resource' from informer's store.
 	// The informer for 'resource' will be created if not exist, but without any event handler.
 	Lister(resource schema.GroupVersionResource) cache.GenericLister
 
+	// Indexer returns a indexer used to index specified 'resource' from informer's store.
+	// The informer for 'resource' will be created if not exist, but without any event handler.
+	Indexer(resource schema.GroupVersionResource) cache.Indexer
+
```

Comment:
Does the two newly added interface can fit your expectation you described before? 

>informermanager uses DynamicClientSet to handle resource event. When we list all nodes and pods, we've got a slice of runtime.object. Then we should use DefaultUnstructuredConverter to convert all objects to v1.Node or v1.Pod, which will use json.Unmarshal for Deserialization. Meanwhile, the cluster status may be reconciled frequently so informermanager may be a little expensive. Furthermore, node and pod lister use protobuf which may be more suitable.

Reply:
These interfaces are designed for indexing pods which are assigned to one node.